### ISSUE TITLE
aws_cloudfront_distribution.main.origin.domain_name specified twice and second is wrong

### ISSUE THREAD

**Issue by colindean** [riboseinc/terraform-aws-s3-cloudfront-website#5]
The `domain_name` property is provided twice:

https://github.com/riboseinc/terraform-aws-s3-cloudfront-website/blob/bfa5677b2a9eb6ff3e925dceb8e241741b345079/cloudfront.tf#L12-L19

This leads to a difficult to troubleshoot problem because everything looks right but it's subtlely wrong, or at least it didn't work for me.

It seems that _either_ line works in my tests, but the second needs a slight modification:

```
domain_name = "${var.fqdn}.s3-website.${data.aws_region.main.name}.amazonaws.com"
```

Note the `.` instead of `-`.

**Comment by colindean** [comment]
On a whim, I tried uncommenting https://github.com/riboseinc/terraform-aws-s3-cloudfront-website/blob/bfa5677b2a9eb6ff3e925dceb8e241741b345079/cloudfront.tf#L18 and it worked, so that must have been added in the meantime. I'll open a PR that uses that instead.

**Comment by ronaldtse** [comment]
@colindean thank you for the report, I admit the double-set is a ridiculous error. 😭 

However, in my tests (our sites), the `domain_name` value is indeed set to something like:
`example.com.s3-website-us-east-1.amazonaws.com`, with the hyphen ("-") instead of the period (".").

Are you seeing something different?

P.S. The bug specified in the note is still not fixed (https://github.com/terraform-providers/terraform-provider-aws/issues/1117)

**Comment by ronaldtse** [comment]
@colindean just a note regarding the uncommenting -- previously, it will work IFF you have already applied once (i.e. the S3 bucket has already been created). It did not work when it the S3 bucket needs to be created.

Did it work in the latter case?

**Comment by colindean** [comment]
> Are you seeing something different?

Yes. `example.com.s3-website.us-east-2.amazonaws.com`. I don't know if that's specific to `us-east-2`, but I doubt it.

Using `domain_name = "${aws_s3_bucket.main.bucket_domain_name}"` seemed to work, too, but…

> It did not work when it the S3 bucket needs to be created.

Ohhhhhhhh yeah, the bucket already existed when I tried _both_ of those previously commented-out lines.

**Comment by ronaldtse** [comment]
> Yes. example.com.s3-website.us-east-2.amazonaws.com. I don't know if that's specific to us-east-2, but I doubt it.

I just created a site yesterday in `us-west-2`, but it gives me the hyphen...?

`www.example.org.s3-website-us-west-2.amazonaws.com`

The problem is the hyphen didn't work for you? I just tried both the "dot" and the "hyphen" and they both seemed to work?

**Comment by colindean** [comment]
http://staging.abstractions.io.s3-website.us-east-2.amazonaws.com succeeds
http://staging.abstractions.io.s3-website-us-east-2.amazonaws.com fails

**Comment by ronaldtse** [comment]
Closed in #6 . Thank you @colindean !
